### PR TITLE
feat(engine+ui): Add Datadog security signals actions

### PR DESCRIPTION
**Pull request by topher-lo**
Closes part of tracker issue #67

Note: still untested. Requires an AWS-Datadog lab environment (with stratus-red-team to generator signals).

- [x] Integration and action form shows up correctly in frontend
dd_integrations

- [x] Confirmed type hints to UI works

image

- [x] Confirm form saves and state shows up in JSON viewer

### CODE REVIEW

File: tracecat/integrations/datadog.py [hunk 1]
```diff
@@ -1,13 +1,178 @@
...
+    description="Get Datadog SIEM security signals. Requires `security_monitoring_signals_read` scope.",
+    secrets=["datadog-security-monitoring"],
+)
+def list_security_signals(
+    query: str | None = None,
+    start: datetime | None = None,
+    end: datetime | None = None,
+    limit: int = 100,
+    region: str = "us1",
+) -> list[dict]:
```
Comment (daryllimyt) [inline tracecat/integrations/datadog.py]:
```suggestion
) -> list[dict[str, Any]]:
```

Comment (daryllimyt) [reply tracecat/integrations/datadog.py]: + from typing import Any

File: tracecat/integrations/datadog.py [hunk 2]
```diff
@@ -1,13 +1,178 @@
...
+        "none",
+        "false_positive",
+        "testing_or_maintenance",
+        "investigated_case_opened",
+        "other",
+    ],
+    archive_comment: str | None = None,
+    archive_reason: str | None = None,
+    region: str = "us1",
+) -> dict:
```
Comment (daryllimyt) [inline tracecat/integrations/datadog.py]: same as above

**Review by daryllimyt — CHANGES_REQUESTED** [review]
Minor typing additions, LGTM. Final check - do the UI form elements show up correctly?

File: tracecat/integrations/datadog.py [hunk 3]
```diff
@@ -1,13 +1,178 @@
...
```
Comment (topher-lo) [inline tracecat/integrations/datadog.py]:
```suggestion
) -> dict[str, Any]:
```

**Pull request merged**